examples/rust/custom_space_view/src/color_coordinates_space_view.rs

```diff
@@ -0,0 +1,248 @@
+fn draw_color_space(
+    ui: &mut egui::Ui,
+    ctx: &mut ViewerContext<'_>,
+    space_view_id: SpaceViewId,
+    scene: &mut TypedScene<ColorCoordinatesSpaceView>,
+    color_at: impl Fn(f32, f32) -> egui::Color32,
+    position_at: impl Fn(egui::Color32) -> (f32, f32),
+) -> egui::Response {
+    // Number of vertices per dimension.
+    // We need at least 6 for hues, and more for smooth 2D areas.
+    // Should always be a multiple of 6 to hit the peak hues in HSV/HSL (every 60°).
+    const N: u32 = 6 * 6;
+
+    let (rect, response) = ui.allocate_exact_size(ui.available_size(), egui::Sense::click());
+
+    if ui.is_rect_visible(rect) {
+        // Background space.
+        let mut mesh = egui::Mesh::default();
+        for xi in 0..=N {
+            for yi in 0..=N {
+                let xt = xi as f32 / (N as f32);
+                let yt = yi as f32 / (N as f32);
+                let color = color_at(xt, yt);
+                let x = egui::lerp(rect.left()..=rect.right(), xt);
+                let y = egui::lerp(rect.bottom()..=rect.top(), yt);
+                mesh.colored_vertex(egui::pos2(x, y), color);
+
+                if xi < N && yi < N {
+                    let x_offset = 1;
+                    let y_offset = N + 1;
+                    let tl = yi * y_offset + xi;
+                    mesh.add_triangle(tl, tl + x_offset, tl + y_offset);
+                    mesh.add_triangle(tl + x_offset, tl + y_offset, tl + y_offset + x_offset);
+                }
+            }
+        }
+        ui.painter().add(egui::Shape::mesh(mesh));
+
+        // Circles for the colors in the scene.
+        for (i, (instance, color)) in scene.parts.colors.iter().enumerate() {
+            let highlight = scene
+                .highlights
+                .entity_highlight(instance.entity_path_hash)
+                .index_highlight(instance.instance_key);
+
+            let (x, y) = position_at(*color);
+            let center = egui::pos2(
+                egui::lerp(rect.left()..=rect.right(), x),
+                egui::lerp(rect.bottom()..=rect.top(), y),
+            );
+
+            // Change color & size depending on whether this instance is selected.
+            let (color, radius) = match (
+                highlight.hover,
+                highlight.selection != SelectionHighlight::None,
+            ) {
+                (HoverHighlight::None, false) => (egui::Color32::BLACK, 2.0),
+                (HoverHighlight::None, true) => (ui.style().visuals.selection.bg_fill, 8.0),
+                (HoverHighlight::Hovered, ..) => (ui.style().visuals.widgets.hovered.bg_fill, 8.0),
+            };
+
+            ui.painter()
+                .add(egui::Shape::circle_filled(center, radius, color));
+
+            let interact = ui.interact(
+                egui::Rect::from_center_size(center, egui::Vec2::splat(radius * 2.0)),
+                ui.id().with(i),
```

Comment:
nit: less risk for id clashes, especially if the user adds another loop painting other things

```suggestion
                ui.id().with(("circle", i)),
```